rtsp.go

```diff
@@ -213,141 +226,141 @@ func (rc *rtspCamera) reconnectClient(codecInfo videoCodec) (err error) {
-	if format.SPS != nil {
-		rc.rawDecoder.decode(format.SPS)
+	if f.SPS != nil {
+		rc.rawDecoder.decode(f.SPS) // nolint:errcheck
 	} else {
 		rc.logger.Warn("no SPS found in H264 format")
 	}
-	if format.PPS != nil {
-		rc.rawDecoder.decode(format.PPS)
+	if f.PPS != nil {
+		rc.rawDecoder.decode(f.PPS) // nolint:errcheck
 	} else {
 		rc.logger.Warn("no PPS found in H264 format")
 	}
 
-	// On packet retreival, turn it into an image, and store it in shared memory
-	rc.client.OnPacketRTP(track, format, func(pkt *rtp.Packet) {
-		// extract access units from RTP packets
-		au, _, err := rtpDec.Decode(pkt)
+	storeImage := func(pkt *rtp.Packet) {
+		au, err := rtpDec.Decode(pkt)
 		if err != nil {
 			if err != rtph264.ErrNonStartingPacketAndNoPrevious && err != rtph264.ErrMorePacketsNeeded {
-				rc.logger.Errorf("error decoding(1) h264 rstp stream %v", err)
+				rc.logger.Errorf("error decoding(1) h264 rstp stream err: %s", err.Error())
 			}
 			return
 		}
 
 		for _, nalu := range au {
-
 			// convert NALUs into RGBA frames
-			lastImage, err := rc.rawDecoder.decode(nalu)
-
+			image, err := rc.rawDecoder.decode(nalu)
 			if err != nil {
-				rc.logger.Error("error decoding(2) h264 rtsp stream  %v", err)
+				rc.logger.Errorf("error decoding(2) h264 rtsp stream  %s", err.Error())
 				return
 			}
-
-			if lastImage != nil {
-				rc.latestFrame.Store(&lastImage)
+			if image != nil {
+				rc.latestFrame.Store(&image)
 			}
 		}
-	})
+	}
+
+	onPacketRTP := func(pkt *rtp.Packet) {
+		storeImage(pkt)
+	}
+
+	_, err = rc.client.Setup(session.BaseURL, media, 0, 0)
+	if err != nil {
+		return errors.Wrapf(err, "when calling RTSP Setup on %s for H264", session.BaseURL)
+	}
+
+	rc.client.OnPacketRTP(media, f, onPacketRTP)
 
 	return nil
 }
 
 // initH265 initializes the H265 decoder and sets up the client to receive H265 packets.
-func (rc *rtspCamera) initH265(tracks media.Medias, baseURL *url.URL) (err error) {
-	var format *formats.H265
+func (rc *rtspCamera) initH265(session *description.Session) (err error) {
+	var f *format.H265
 
-	track := tracks.FindFormat(&format)
-	if track == nil {
+	media := session.FindFormat(&f)
+	if media == nil {
 		rc.logger.Warn("tracks available")
-		for _, x := range tracks {
+		for _, x := range session.Medias {
 			rc.logger.Warnf("\t %v", x)
 		}
 		return errors.New("h265 track not found")
 	}
 
-	_, err = rc.client.Setup(track, baseURL, 0, 0)
-	if err != nil {
-		return err
-	}
-
-	rtpDec, err := format.CreateDecoder2()
+	rtpDec, err := f.CreateDecoder()
 	if err != nil {
-		rc.logger.Errorf("error creating H265 decoder %v", err)
-		return err
+		return errors.Wrap(err, "creating H265 RTP decoder")
 	}
 
 	rc.rawDecoder, err = newH265Decoder()
 	if err != nil {
-		return err
+		return errors.Wrap(err, "creating H265 raw decoder")
 	}
 
 	// For H.265, handle VPS, SPS, and PPS
-	if format.VPS != nil {
-		rc.rawDecoder.decode(format.VPS)
+	if f.VPS != nil {
+		rc.rawDecoder.decode(f.VPS) // nolint:errcheck
```

Comment:
Same as in your other pr, were we just not lining and never checking the error? Okay to address in the other pr.

Reply:
Yes. We are not running lint in CI (we should change that) and we were not checking the error

Reply:
https://viam.atlassian.net/browse/RSDK-7315